src/cortxfs/cortxfs_ops.c

```diff
@@ -78,58 +78,15 @@ int cfs_set_stat(struct kvnode *node)
 	return rc;
 }
 
-static inline int __cfs_getattr(struct cfs_fs *cfs_fs, const cfs_cred_t *cred,
-			 const cfs_ino_t *ino, struct stat *bufstat)
+static inline int __cfs_setattr(struct cfs_fh *fh, cfs_cred_t *cred,
+				struct stat *setstat, int statflag)
 {
-	int rc;
-	struct stat *stat = NULL;
-	struct cfs_fh *fh = NULL;
-
-	dassert(cfs_fs && cred && ino && bufstat);
-
-	/* TODO:Temp_FH_op - to be removed
-	 * Should get rid of creating and destroying FH operation in this
-	 * API when caller pass the valid FH instead of inode number
-	 */
-	RC_WRAP_LABEL(rc, out, cfs_fh_from_ino, cfs_fs, ino, &fh);
-
-	stat = cfs_fh_stat(fh);
-
-	memcpy(bufstat, stat, sizeof(struct stat));
-out:
-	if (fh != NULL) {
-		cfs_fh_destroy_and_dump_stat(fh);
-	}
-
-	log_debug("ino=%d rc=%d", (int)bufstat->st_ino, rc);
-	return rc;
-}
-
-int cfs_getattr(struct cfs_fs *cfs_fs, const cfs_cred_t *cred,
```

Comment:
In my opinion we should have this function. This will simply call cfs_fh_stat.
"setattr" and "getattr" are 2 sides of coin, so keep it.

Reply:
I agree @sachinpunadikar .Probably ` cfs_getattr ( )` could be a wrapper over `cfs_fh_stat()`, and it should return a integer value if the caller was able to fetch the attributes. 
Again this might result in one more review cycle and testing, which could atleast take couple of days more at a minimum.
@sachinpunadikar : Please let us know your views on whether we should continue with these changes.
Also,
There are few places in the code where we return a stat pointer after call to cfs_fh_stat (), and use that pointer subsequently.
It's possible that that pointer could be NULL (due to any internal error etc). A rc check could avoid this.
Again, this is future work, we can definitely skip this for now I feel.

Reply:
for logical conclusion on this, better to have cfs_getattr. I am fine with some more time spent on this.

Reply:
> I agree @sachinpunadikar .Probably ` cfs_getattr ( )` could be a wrapper over `cfs_fh_stat()`, and it should return a integer value if the caller was able to fetch the attributes.
Yes, this is doable and we can check for error value returned.
> Again this might result in one more review cycle and testing, which could atleast take couple of days more at a minimum.
> @sachinpunadikar : Please let us know your views on whether we should continue with these changes.
Yes, this is necessary.
> Also,
> There are few places in the code where we return a stat pointer after call to cfs_fh_stat (), and use that pointer subsequently.
> It's possible that that pointer could be NULL (due to any internal error etc). A rc check could avoid this.
No, not possible because we first check file handle and if it is sane, only then we call cfs_fh_stat.
> Again, this is future work, we can definitely skip this for now I feel.